providers/logingov.go

```diff
@@ -88,66 +90,71 @@ type loginGovCustomClaims struct {
 }
 
 // checkNonce checks the nonce in the id_token
-func checkNonce(idToken string, p *LoginGovProvider) (err error) {
+func checkNonce(ctx context.Context, idToken string, p *LoginGovProvider) error {
 	token, err := jwt.ParseWithClaims(idToken, &loginGovCustomClaims{}, func(token *jwt.Token) (interface{}, error) {
-		resp, myerr := http.Get(p.PubJWKURL.String())
-		if myerr != nil {
-			return nil, myerr
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, p.PubJWKURL.String(), nil)
+		if err != nil {
+			return nil, err
+		}
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return nil, err
 		}
-		if resp.StatusCode != 200 {
-			myerr = fmt.Errorf("got %d from %q", resp.StatusCode, p.PubJWKURL.String())
-			return nil, myerr
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("got %d from %q", resp.StatusCode, p.PubJWKURL.String())
 		}
-		body, myerr := ioutil.ReadAll(resp.Body)
+		body, err := ioutil.ReadAll(resp.Body)
 		resp.Body.Close()
-		if myerr != nil {
-			return nil, myerr
+		if err != nil {
+			return nil, err
 		}
 
 		var pubkeys jose.JSONWebKeySet
-		myerr = json.Unmarshal(body, &pubkeys)
-		if myerr != nil {
-			return nil, myerr
+		err = json.Unmarshal(body, &pubkeys)
+		if err != nil {
+			return nil, err
+		}
+		if len(pubkeys.Keys) == 0 {
+			return nil, fmt.Errorf("no keys in jwt: pubkeys=%v", pubkeys)
```

Comment:
```suggestion
			return nil, fmt.Errorf("no keys in jwks")
```

Reply:
`jwks` How do you mean?
`jwts`?, `jwt keys` ?